kafka-rest/src/test/java/io/confluent/kafkarest/v2/KafkaConsumerManagerTest.java

```diff
@@ -381,27 +383,56 @@ public void onCompletion(List<TopicPartitionOffset> offsets, Exception e) {
         consumerManager.deleteConsumer(groupName, consumer.cid());
     }
 
-    @Test
-    public void testBackoffMsControlsPollCalls() throws Exception {
-        bootstrapConsumer(consumer);
-        consumerManager.readRecords(groupName, consumer.cid(), BinaryKafkaConsumerState.class, -1, Long.MAX_VALUE,
-                new ConsumerReadCallback<ByteString, ByteString>() {
-                    @Override
-                    public void onCompletion(
-                        List<ConsumerRecord<ByteString, ByteString>> records, Exception e) {
-                        actualException = e;
-                        actualRecords = records;
-                        sawCallback = true;
-                    }
-                });
-
-        // backoff is 250
-        Thread.sleep(100);
-        // backoff should be in place right now. the read task should be delayed and re-ran until the max.bytes or timeout is hit
-        assertEquals(1, consumerManager.delayedReadTasks.size());
-        Thread.sleep(100);
-        assertEquals(1, consumerManager.delayedReadTasks.size());
+  @Test
+  public void testBackoffMsControlsPollCalls() throws Exception {
+    long timeoutMillis = 1000;
+    long backoffMillis = 100;
+    Properties props = setUpProperties();
+    props.put(KafkaRestConfig.CONSUMER_REQUEST_TIMEOUT_MS_CONFIG, String.valueOf(timeoutMillis));
+    props.put(KafkaRestConfig.CONSUMER_ITERATOR_BACKOFF_MS_CONFIG, String.valueOf(backoffMillis));
+    config = new KafkaRestConfig(props, new SystemTime());
+    consumerManager = new KafkaConsumerManager(config, consumerFactory);
+    consumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST, groupName);
+    bootstrapConsumer(consumer);
+    ArrayList<Long> pollTimestamps = new ArrayList<>();
```

Comment:
nit: I'm not sure on which thread is this mutated, but just to be sure, let's use `CopyOnWriteArrayList`.

Reply:
Done.